clients/tabby-chat-panel/src/react.ts

```diff
@@ -6,21 +6,25 @@ import { createClient, createServer } from './index'
 
 function useClient(iframeRef: RefObject<HTMLIFrameElement>) {
   return useMemo(() => {
-    if (iframeRef.current)
+    if (iframeRef.current) {
       return createClient(iframeRef.current)
+    }
   }, [iframeRef.current])
 }
 
 function useServer(api: Api) {
   const [isInIframe, setIsInIframe] = useState(false)
+  let isCreated = false
 
   useEffect(() => {
     setIsInIframe(window.self !== window.top)
   }, [])
 
   return useMemo(() => {
-    if (isInIframe)
+    if (isInIframe && !isCreated) {
+      isCreated = true
       return createServer(api)
+    }
   }, [isInIframe])
```

Comment:
i think this can be simplified with:
1. create a ref = useRef
2. in useEffect, when window.self !== window.top, set ref.current = createServer

Reply:
updated